SignalServiceKit/src/Loki/API/FileServerAPI.swift

```
@@ -52,6 +53,37 @@ public final class FileServerAPI : DotNetAPI {
         }
     }
     
+    @objc(downloadProfilePicture:)
+    public static func objc_downloadProfilePicture(_ downloadURL: String) -> AnyPromise {
+        return AnyPromise.from(downloadAttachment(downloadURL))
+    }
+    
+    // MARK: Attachment Download
+    @objc(downloadAttachment:)
+    public static func objc_downloadAttachment(_ downloadURL: String) -> AnyPromise {
+        return AnyPromise.from(downloadAttachment(downloadURL))
+    }
+    
+    public static func downloadAttachment(_ downloadURL: String) -> Promise<Data> {
+        var error: NSError?
+        var url = downloadURL
+        if downloadURL.contains(fileStaticServer) {
+            url = downloadURL.replacingOccurrences(of: fileStaticServer, with: "\(server)/loki/v1")
+        }
+        let request = AFHTTPRequestSerializer().request(withMethod: "GET", urlString: url, parameters: nil, error: &error)
+        if let error = error {
+            print("[Loki] Couldn't download attachment due to error: \(error).")
+            return Promise(error: error)
+        }
+        return OnionRequestAPI.sendOnionRequest(request, to: server, using: fileServerPublicKey, isJSONRequired: false).map2 { json in
+            guard let body = json["body"] as? JSON, let dataArray = body["data"] as? [UInt8] else {
+                print("[Loki] Couldn't download attachment.")
+                return Data()
```

Let's throw an error here to indicate that the JSON returned was invalid.